src/azure-cli/azure/cli/command_modules/network/private_link_resource_and_endpoint_connections/resource_providers/__init__.py

```diff
@@ -0,0 +1,182 @@
+                                            name, approval_description=None):
+        return
+
+    @abc.abstractmethod
+    def reject_private_endpoint_connection(self, cmd, resource_group_name, service_name,
+                                           name, rejection_description=None):
+        return
+
+    @abc.abstractmethod
+    def remove_private_endpoint_connection(self, cmd, resource_group_name, service_name, name):
+        return
+
+    @abc.abstractmethod
+    def show_private_endpoint_connection(self, cmd, resource_group_name, service_name, name):
+        return
+
+
+class GeneralPrivateEndpointClient(PrivateEndpointClient):
+
+    def __init__(self, rp, api_version, support_list):
+        self.rp = rp
+        self.api_version = api_version
+        self.support_list = support_list
+
+    def _update_private_endpoint_connection_status(self, cmd, resource_group_name,
+                                                   service_name, private_endpoint_connection_name,
+                                                   is_approved=True, description=None):
+
+        private_endpoint_connection = self.show_private_endpoint_connection(cmd,
+                                                                            resource_group_name,
+                                                                            service_name,
+                                                                            private_endpoint_connection_name)
+
+        new_status = "Approved" if is_approved else "Rejected"
+        private_endpoint_connection['properties']['privateLinkServiceConnectionState']['status'] = new_status
+        private_endpoint_connection['properties']['privateLinkServiceConnectionState']['description'] = description
+
+        url = _build_connection_url_endpoint(resource_group_name,
+                                             self.rp,
+                                             service_name,
+                                             private_endpoint_connection_name,
+                                             self.api_version)
+        r = send_raw_request(cmd.cli_ctx, 'put', url, body=json.dumps(private_endpoint_connection))
+        query_counts = RETRY_MAX
+        while query_counts:
+            time.sleep(RETRY_INTERVAL)
+            query_counts -= 1
+            private_endpoint_connection = self.show_private_endpoint_connection(cmd,
+                                                                                resource_group_name,
+                                                                                service_name,
+                                                                                private_endpoint_connection_name)
+            if private_endpoint_connection['properties'].get('provisioningState'):
+                if private_endpoint_connection['properties']['provisioningState'] == "Succeeded":
+                    return private_endpoint_connection
```

Comment:
Maybe we can extract this part and put it into core, so that `az rest` can also benefit from it by using `az rest --wait-async`.

Reply:
the show method is specific to each get rest call. `wait command` requires user's input to identify the resource. So I think it's not suitable to have a generic way.

Reply:
Made a demo:

```py
def send_raw_request_wait(*args, timeout=3600, expected_status_code=None,
                          body_query=None, expected_body_query_result=None, **kwargs):
    RETRY_MAX = 20
    RETRY_INTERVAL = 10
    query_counts = RETRY_MAX
    import time
    import jmespath
    # src/azure-cli-core/azure/cli/core/commands/arm.py:676
    while query_counts:
        time.sleep(RETRY_INTERVAL)
        query_counts -= 1
        response = send_raw_request(*args, **kwargs)
        if body_query and jmespath.search(body_query, response) != expected_body_query_result:
            continue
        if expected_status_code:
            raise NotImplementedError()
    raise Exception("Max retry count reached.")
```

But we can consider it later. No hurry.